### PR TITLE
make sure that authorization policies XDS is never in on-demand mode

### PR DESCRIPTION
**Pull request by yuval-k**
As auth policies have deny rules, they must all be present in the ztunnel, even in on demand mode.

The PR changes the ztunnel behavior to never issue an on-demand XDS for auth policies

### CODE REVIEW

File: src/xds/client.rs [hunk 1]
```diff
@@ -141,7 +144,7 @@ impl<T: 'static + prost::Message + Default> RawHandler for HandlerWrapper<T> {
                 state.add_resource(key.type_url, key.name);
                 r
             })
-            .map(|raw| decode_proto::<T>(raw).unwrap())
+            .map(|raw| decode_proto::<T>(&raw).unwrap())
```
Comment (howardjohn) [inline src/xds/client.rs:147]: why this change?  I think this means we do a clone() on the full proto which we shouldn't need?

Comment (yuval-k) [reply src/xds/client.rs:147]: we don't need this changed now, i left there by mistake from another change I'm working on. s

It should matter in terms of `clone()`ing though. You can see that `decode_proto` doesn't do any additional clone() operations

If you prefer i can revert it from here and add it in the next PR I'm working on.

Comment (yuval-k) [reply src/xds/client.rs:147]: sorry, `decode_proto` does clone the name now; i'll revert it from here and add it back in context so it will make more sense

**Review by howardjohn — APPROVED** [review]

File: src/xds/client.rs (hunk 1 above)
Comment (howardjohn) [reply src/xds/client.rs:147]: nvm, misread it

Comment (howardjohn) [reply src/xds/client.rs:147]: I put hold incase you want to fix, feel free to remove

**Comment by yuval-k** [comment]
fixed and removed the hold
